target/ppc/int_helper.c

```diff
@@ -963,33 +964,39 @@ VCT(sxs, cvtsdsw, s32)
     }
 
 
-#define GER(bits, rank, el)                                             \
+#define GER_MULT                                                        \
     do {                                                                \
-        va = a->VsrN((rank*i + k));                                     \
+        va = (a->VsrB((i*8 + k)/2) >> ((k+1) % 2)*4) & 0xF;             \
```

Comment:
I still think it would be clearer, even using GER_MULT, to use the GET_Vsr* macros to distinguish between N/B/H - but then it would be a single GER_MULT, with varying GET_Vsr*.

I understand this what you're trying to do is to have a single XVIGER that can be reused by all of these instructions and also all of the floating point ger instructions as well?

To decide, it would be good for us (and @mferst) to review what the XVIGER code would look like, with all the scenarios handled. What do you think?

BTW, for floating point, will we also need to redefine psum (not as int64_t)?